client/src/GuessGame/GuessGame.js

```diff
@@ -0,0 +1,134 @@
+import React, { useEffect, useState } from 'react';
+import chroma from 'chroma-js';
+import countryList from './countryList';
+import Datamap from './DataMap';
+import './GuessGame.css';
+
+
+//Start the game
+const GuessGame = () => {
+  const maxScore = countryList.length;
+  const [guessCounter, setGuessCounter] = useState(0);
+  const [currCountry, setCurrCountry] = useState({});
+  const [data, setData] = useState({});
+  const [win, setWin] = useState(false);
+
+  useEffect(() => {
+    getRandCountry();
+
+    return () => {};
+  }, []);
+
+  //Get a random integer to grab random country.
+  const getRandInteger = (min, max) => {
+    return Math.floor(Math.random() * (max - min + 1) + min);
+  };
+
+  // Using random integer get the random country
+  const getRandCountry = () => {
+    const countryNum = getRandInteger(0, maxScore - 1);
+    setCurrCountry(countryList[countryNum]);
+    console.log('Random country: ', currCountry, countryList[countryNum]);
+  };
+
+  //Get the distance via euclidean maths
+  const distance = (srcLat, srcLong) => {
+    return Math.sqrt(
+      Math.pow(currCountry['longitude'] - srcLong, 2) +
+        Math.pow(currCountry['latitude'] - srcLat, 2)
+    );
+  };
+
+  // For the cheaters out there.
+  console.log(currCountry['country']);
+
+  // User submitted answer
+  const submitGuess = (userInput) => {
+    if (userInput.toLowerCase() === currCountry['country'].toLowerCase()) {
+      setData({ [currCountry['alpha3']]: 'green' });
+      setGuessCounter(guessCounter + 1);
+      setWin(true);
+    } else {
+      const guessCode = countryList.find((countryObj) => {
+        return countryObj.country.toLowerCase() === userInput.toLowerCase();
+      });
+      if (!guessCode) {
+        return;
+      }
+      const maxDist = 275;
+      const dist = distance(guessCode['latitude'], guessCode['longitude']);
+      const gradient = chroma.scale(['red','blue']);
+
+      const color = gradient( dist / maxDist ).hex();
+
+      // data[guessCode['alpha3']] = color;
+      setData({ [guessCode['alpha3']]: color });
+      setGuessCounter(guessCounter + 1);
+
+      // console.log(
+      //   ' Lat: ' + guessCode['latitude'] + ' Long: ' + guessCode['longitude']
+      // );
+      // console.log(dist);
+    }
+  };
+
+  //
+  return (
```

Comment:
Nice execution of utilizing your HTML code into JSX to have it work with React. I personally struggle with integrating previous HTML logic to React but its cool to see it be possible!